pkg/pillar/cmd/zedagent/parseedgeview.go

```diff
@@ -0,0 +1,278 @@
+	if err != nil {
+		log.Errorf("json unmarshal algo error: %v", err)
+		return err
+	}
+
+	if jalgo.Alg != types.EdgeviewJWTAlgo || jalgo.Typ != types.EdgeviewJWTType {
+		err := fmt.Errorf("jwt algo incorrect: %v", jalgo)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	method := jwt.GetSigningMethod(jalgo.Alg)
+	err = method.Verify(strings.Join(params[0:2], "."), params[2], ecdsaKey)
+	if err != nil {
+		log.Errorf("verify jwt failed: %v", err)
+		return err
+	}
+	log.Tracef("jwt verify ok")
+
+	return nil
+}
+
+func addEvFiles(evConfig types.EdgeviewConfig, params []string) error {
+	tokenData, err := base64.RawURLEncoding.DecodeString(params[1])
+	if err != nil {
+		log.Errorf("base64 decode jwt error: %v", err)
+		return err
+	}
+
+	var jdata types.EvjwtInfo
+	err = json.Unmarshal(tokenData, &jdata)
+	if err != nil {
+		log.Errorf("json unmarshal jwt error: %v", err)
+		return err
+	}
+
+	if _, err = os.Stat(types.EdgeviewPath); os.IsNotExist(err) {
+		if err := os.MkdirAll(types.EdgeviewPath, 0755); err != nil {
+			log.Errorf("failed to create dir: %v", err)
+			return err
+		}
+	}
+
+	// check devUUID
+	if devUUID.String() != jdata.Sub {
+		err := fmt.Errorf("jwt sub does not match devUUID: %s", jdata.Sub)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// check the expiration time
+	nowSec := uint64(time.Now().Unix())
+	if nowSec > jdata.Exp {
+		err := fmt.Errorf("jwt already expired: %d", jdata.Exp)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// create jwt token file
+	f, err := ioutil.TempFile(types.EdgeviewPath, "Edgeview-Config")
+	if err != nil {
+		log.Errorf("file create failed: %v", err)
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(types.EdgeViewJwtPrefix + evConfig.JWToken + "\n")
+	if err != nil {
+		log.Errorf("file write failed: %v", err)
+		return err
+	}
+
+	_, err = f.WriteString(types.EdgeViewExpPrefix + strconv.FormatUint(jdata.Exp, 10) + "\n")
+	if err != nil {
+		log.Errorf("file write failed: %v", err)
+		return err
+	}
+
+	// write the dispater certs
+	var certs []byte
+	for _, c := range evConfig.DispCertPEM {
+		certs = append(certs, c...)
+	}
+	if len(certs) > 0 {
+		_, err = f.WriteString(types.EdgeViewCertPreifx + string(certs) + "\n")
+		if err != nil {
+			log.Errorf("cert file write failed: %v", err)
+			return err
+		}
+	}
+
+	// write device policy
+	devbytes, err := json.Marshal(evConfig.DevPolicy)
+	if err != nil {
+		log.Errorf("json marshal failed: %v", err)
+		return err
+	}
+
+	_, err = f.WriteString(types.EdgeViewDevPolicyPreifx + string(devbytes) + "\n")
+	if err != nil {
+		log.Errorf("file write failed: %v", err)
+		return err
+	}
+
+	// write app policy
+	appbytes, err := json.Marshal(evConfig.AppPolicy)
+	if err != nil {
+		log.Errorf("json marshal failed: %v", err)
+		return err
+	}
+
+	_, err = f.WriteString(types.EdgeViewAppPolicyPreifx + string(appbytes) + "\n")
+	if err != nil {
+		log.Errorf("file write failed: %v", err)
+		return err
+	}
+
```

Comment:
Shouldn't you do an explicit f.Close() before the Rename? Seems odd to keep the file open while renaming.
(Note that the fact that defer would also result in a second f.Close() is just a no-op.)